internal/provider/resource_gitlab_group_project_file_template.go

```diff
@@ -0,0 +1,133 @@
+
+-> This resource requires a GitLab Enterprise instance with a Premium license.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/groups.html#update-group)`,
+
+		// Since this resource updates an in-place resource, the update method is the same as the create method
+		CreateContext: resourceGitLabGroupProjectFileTemplateCreateOrUpdate,
+		UpdateContext: resourceGitLabGroupProjectFileTemplateCreateOrUpdate,
+		ReadContext:   resourceGitLabGroupProjectFileTemplateRead,
+		DeleteContext: resourceGitLabGroupProjectFileTemplateDelete,
+		// Since this resource updates an in-place resource, importing doesn't make much sense. Simply add the resource
+		// to the config and terraform will overwrite what's already in place and manage it from there.
+		Schema: map[string]*schema.Schema{
+			"group_id": {
+				Description: `The ID of the group that will use the file template project. This group must be the direct
+                parent of the project defined by project_id`,
+				Type: schema.TypeInt,
+
+				// Even though there is no traditional resource to create, leave "ForceNew" as "true" so that if someone
+				// changes a configuration to a different group, the old group gets "deleted" (updated to have a value
+				// of 0).
+				ForceNew: true,
+				Required: true,
+			},
+			"file_template_project_id": {
+				Description: `The ID of the project that will be used for file templates. This project must be the direct
+				child of the project defined by the group_id`,
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+		},
+	}
+})
+
+func resourceGitLabGroupProjectFileTemplateRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	groupID := d.Get("group_id").(int)
+	group, _, err := client.Groups.GetGroup(groupID, nil, gitlab.WithContext(ctx))
+	if err != nil {
+		if is404(err) {
+			log.Printf("[DEBUG] gitlab group %d not found so removing from state", groupID)
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+	if group.MarkedForDeletionOn != nil {
+		log.Printf("[DEBUG] gitlab group %s is marked for deletion", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(fmt.Sprintf("%d", group.ID))
+	d.Set("file_template_project_id", group.FileTemplateProjectID)
+
+	return nil
+}
+
+func resourceGitLabGroupProjectFileTemplateCreateOrUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	groupID := d.Get("group_id").(int)
+
+	// Creating the resource means updating the existing group to link the project to the group.
+	options := &gitlab.UpdateGroupOptions{}
+	if d.HasChanges("file_template_project_id") {
+		options.FileTemplateProjectID = gitlab.Int(d.Get("file_template_project_id").(int))
+	}
+
+	_, _, err := client.Groups.UpdateGroup(groupID, options)
+	if err != nil {
+		return diag.Errorf("could not update group after creation %d: %s", groupID, err)
```

Comment:
```suggestion
		return diag.Errorf("unable to update group %d with `file_template_project_id` set to %d: %s", groupID, projectID, err)
```

This also requires to set the `projectID` to `d.Get(...)` ...